public/app/plugins/datasource/prometheus/datasource.ts

```diff
@@ -365,6 +366,10 @@ export class PrometheusDatasource implements DataSourceApi<PromQuery> {
             };
 
             if (annotation.useValueForTime) {
+              if (dupCheck[value[1]]) {
+                continue;
+              }
+              dupCheck[value[1]] = true;
```

Comment:
looks like value[1] is used in a at least 3 places in this loop, maybe extract value[1] and to variables?

Reply:
Of course, I did!